docs/pipelines.md

```diff
@@ -116,75 +117,115 @@ cohortextractor run dummy generate_cohorts expectations
 	* choose a name, for example the name of the repo
 	* select a database to run against (either dummy data, real data, or a small sample of the real data)
-	* select you repo and branch
+	* select the repo and branch that you want to run actions with
 	* click `Submit`.
 *  **Select actions** to run:
-	* select a many actions as you like by clicking the `Run` buttons
-	* any actions with dependencies will also be run, unless they have already been run
+	* select the actions you want to run by clicking the `Run` buttons
+	* if any of these actions have dependencies then they will also be run, unless their outputs already exist
 	* dependencies can be viewed by clicking the `Needs` button
 	* you can force dependencies to be run by clicking `Force run dependencies`, even if those actions have already been run
 	* when you're ready, click `Submit`.
 
 The workspace is available at `https://jobs.opensafely.org/<WORKSPACE_NAME>/`.
 You can view the progress of these actions by click the `Logs` button from the workspace, or going to `https://jobs.opensafely.org/<WORKSPACE_NAME>/logs`.
 
+<details>
+  <summary>Click here for information on the exact steps that occur when each job is run on the server</summary>
+  
+  1. Create a new, empty temporary directory for the job
+  2. Copy in all files on the selected branch
+  3. Run the job
+  4. Find all the files matching the specified output patterns
+  5. Copy these files into the highly sensitive and moderately sensitive holding directories on the server
+  6. Save the log files for the job into the `metadata/` directory
+  7. Delete the temporary directory
+</details>
+
+
+### Accessing the outputs
+
 If the actions run successfully, the outputs will be created on the server.
 For security reasons, they will be in a different directory than if you had run locally.
 
-For those with Level 4 access:
+No data should ever be published from the Level 3 server. Access is only for permitted users, for the purpose of debugging problems in the secure environment.
+
+#### Level 4 access
 
 * For the TPP backend, outputs labelled `moderately_sensitive` in the `project.yaml` will be saved in `D:/Level4Files/workspaces/<NAME_OF_YOUR_WORKSPACE>`
-* outputs labelled `highly_sensitive` are not visible
+* Outputs labelled `highly_sensitive` are not visible
 
 These outputs can be [reviewed on the server](workflow-check-disclosivity.md) and released via GitHub if they are deemed non-disclosive.
 
-For those with Level 3 access:
-
-* all outputs will be saved in `E:/high_privacy/workspaces/<WORKSPACE_NAME>`
-* there's also a directory called `metadata`, containing log files for each action e.g. generate_chorts.log, run_model.log
+#### Level 3 access
 
-No data should ever be published from the Level 3 server. Access is only for permitted users, for the purpose of debugging problems in the secure environment.
+* All outputs will be saved in `E:/high_privacy/workspaces/<WORKSPACE_NAME>`
```

Comment:
I think we should keep something _like_ "For those with Level 3 access" or similar -- anything to indicate that most people should never expect to do this and/or pay any attention to this section. Don't want to encourage people to think about using L3.

Reply:
Yes good point. I've re-written level 3 info as a subsection.